src/sgp/FileMan.cc

```diff
@@ -672,70 +634,72 @@ FILE* FileMan::openForReadingCaseInsensitive(const std::string &folderPath, cons
 	return NULL;
 }
 
-/**
- * Find all files with the given extension in the given directory.
- * @param dirPath Path to the directory
- * @param extension Extension with dot (e.g. ".txt")
- * @param caseIncensitive When True, do case-insensitive search even of case-sensitive file-systems. * * @return List of paths (dir + filename). */
 std::vector<std::string>
 FindFilesInDir(const std::string &dirPath,
 		const std::string &ext,
 		bool caseIncensitive,
 		bool returnOnlyNames,
 		bool sortResults)
 {
-	std::string ext_copy = ext;
-	if(caseIncensitive)
-	{
-		std::transform(ext_copy.begin(), ext_copy.end(), ext_copy.begin(), ::tolower);
-	}
-
-	std::vector<std::string> paths;
-	boost::filesystem::path path(dirPath);
-	boost::filesystem::directory_iterator end;
-	for(boost::filesystem::directory_iterator it(path); it != end; it++)
+	std::vector<std::string> ret;
+	std::vector<std::string> paths = FindAllFilesInDir(dirPath, sortResults);
+	for (std::string& path : paths)
 	{
-		if(boost::filesystem::is_regular_file(it->status()))
+		RustPointer<char> path_ext(Path_extension(path.c_str()));
```

Comment:
Ideally we should do most of this logic in rust already, but for the time being it think it is ok...

Reply:
Yeah, I plan to rethink paths when the non-boost filesystem stuff is moved to rust.